### ISSUE TITLE
Import Excel Exception Could not load file or assembly 'Microsoft.Office.Interop.Excel, Version=15.0.0.0

### ISSUE THREAD

**Issue by Kulkul7536** [DynamoDS/Dynamo#11632]
Hi,

I'm not able to Read Excel from Dynamo Sandbox 2.10 or 2.11. I'm getting this error:
![image](https://user-images.githubusercontent.com/26419681/115291085-5bde5b80-a165-11eb-903f-d08dc4427cf5.png)
I downloaded latest v15 Microsoft.Office.Interop.Excel.dll from Nuget and pasted in DynamoSandbox folder but still the same issue.

Thanks & Regards,

**Comment by Kulkul7536** [comment]
Hi,

I'm not able to Read Excel from Dynamo Sandbox 2.10 or 2.11. I'm getting this error:
![image](https://user-images.githubusercontent.com/26419681/115291085-5bde5b80-a165-11eb-903f-d08dc4427cf5.png)
I downloaded latest v15 Microsoft.Office.Interop.Excel.dll from Nuget and pasted in DynamoSandbox folder but still the same issue.

Thanks & Regards,

**Comment by mjkkirschner** [comment]
Hi @Kulkul7536 - I think theres a few things going on.

1. can you try installing  https://www.microsoft.com/en-us/download/details.aspx?id=3508
2. what version of excel do you have installed? Can you try repairing the install, and making sure you install interop assemblies with it - it may be an option during install.
3. checkout hints here: https://github.com/DynamoDS/Dynamo/wiki/How-to-Utilize-Dynamo-Builds#dynamo-excel-nodes-issues
4. when you copy the dll from nuget - make sure you unblock it. - I don't have high faith in this solution.

**Comment by Kulkul7536** [comment]
Hi @mjkkirschner 

Thanks for replying,

1. Yes i did try installing https://www.microsoft.com/en-us/download/details.aspx?id=3508. Still the same error.
2. Office 2013.
3. I tried to repair excel like shown on the link but i din't find any dll files in C:\Windows\Assemblies...
4. Not sure what you mean by copy the dll from nuget?

Best Regards,
~Kulkul

**Comment by mjkkirschner** [comment]
Hi @Kulkul7536 you said:
```
I downloaded latest v15 Microsoft.Office.Interop.Excel.dll from Nuget and pasted in DynamoSandbox folder but still the same issue.

```

I don't think that is a great solution because:
* the dll should come with excel, if it's not installed, there might be other missing components as well.
* the dll might be blocked, you downloaded it from nuget's website I guess, so it might have "mark of the web" - for example: https://stackoverflow.com/questions/4103683/problem-unblocking-assemblies-in-windows-7-home-premium/4103806

please see this thread regarding 2013:
https://social.msdn.microsoft.com/Forums/sqlserver/en-US/3c74d31d-9c2a-49a4-accf-135267f11984/where-to-find-office-2013-pias-for-install?forum=vsto

it sounds like there is a particular checkbox during install to look for.

**Comment by Kulkul7536** [comment]
Hi @mjkkirschner 

I downloaded v15 because of the exception, it requires v15. Another thing i noticed Excel node works fine with Revit 2019-2021. I get this excel error with Revit 2022, DynamoSandbox 2.10 and DynamoSandbox 2.11. I hope there has been no changes in the development in newer versions?

When i click on Change i get these checkboxes:
![image](https://user-images.githubusercontent.com/26419681/115955945-dfec6680-a50a-11eb-87dd-95ad113763c6.png)
And when i click on Add/Remove Features i get these:
![image](https://user-images.githubusercontent.com/26419681/115955973-26da5c00-a50b-11eb-962a-bf7a938dec5d.png)

**Comment by mjkkirschner** [comment]
Yes, that seems to be the option .Net Programmability Support, is it installed?

**Comment by Kulkul7536** [comment]
Yes it is installed but i didn't find any assembly:
![image](https://user-images.githubusercontent.com/26419681/116002476-f679e700-a60a-11eb-9468-1288c96a3e49.png)
And when i run dynamo the same exception appears.

**Comment by Kulkul7536** [comment]
@mjkkirschner any other workaround?

**Comment by mjkkirschner** [comment]
@Kulkul7536 we just saw this internally on a system, and a restart and excel update fixed it. Maybe give that a shot?

**Comment by Kulkul7536** [comment]
Thanks @mjkkirschner nah restarting didn't fix.
Excel Update - How can I update excel alone?
